src/ext/java/org/opentripplanner/ext/fares/model/FareLegRule.java

```diff
@@ -10,18 +11,20 @@ public record FareLegRule(
   @Nullable String fromAreaId,
   @Nullable String toAreadId,
   @Nullable FareDistance fareDistance,
-  @Nonnull FareProduct fareProduct
+  @Nonnull ArrayList<FareProduct> fareProducts,
+  @Nonnull String feedId
 ) {
   public FareLegRule(
     String legGroupId,
     String networkId,
     String fromAreaId,
     String toAreaId,
-    FareProduct fareProduct
+    ArrayList<FareProduct> fareProducts,
+    String feedId
```

Comment:
Rather than passing the feed id, I think we should give `FareLegRule` an id of type `FeedScopedId`. We don't really expose the id but it feels more elegant to do it that way.

It should be mandatory argument of `FareLegRule.of()`.